Doctrine/Orm/Filter/DateFilter.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/*
+ * This file is part of the DunglasApiBundle package.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Dunglas\ApiBundle\Doctrine\Orm\Filter;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ORM\QueryBuilder;
+use Dunglas\ApiBundle\Api\ResourceInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Filters the collection by date intervals.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ * @author Théo FIDRY <theo.fidry@gmail.com>
+ */
+class DateFilter extends AbstractFilter
+{
+    const PARAMETER_BEFORE = 'before';
+    const PARAMETER_AFTER = 'after';
+    const EXCLUDE_NULL = true;
+
+    /**
+     * @var array
+     */
+    private static $doctrineDateTypes = [
+        'date' => true,
+        'datetime' => true,
+        'datetimetz' => true,
+        'time' => true,
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder, Request $request)
+    {
+        $fieldNames = $this->getDateFieldNames($resource);
+
+        foreach ($this->extractProperties($request) as $property => $values) {
+            // Expect $values to be an array having the period as keys and the date value as values
+            if (!isset($fieldNames[$property]) || !is_array($values) || !$this->isPropertyEnabled($property)) {
+                continue;
+            }
+
+            if (isset($values[self::PARAMETER_BEFORE])) {
+                $this->filterByDate($queryBuilder, $property, $values[self::PARAMETER_BEFORE], self::PARAMETER_BEFORE);
+            }
+
+            if (isset($values[self::PARAMETER_AFTER])) {
+                $this->filterByDate($queryBuilder, $property, $values[self::PARAMETER_AFTER], self::PARAMETER_AFTER);
+            }
+
+            if (self::NULL_EXCLUDED === $this->properties) {
+                $queryBuilder->andWhere(sprintf('o.%s IS NOT NULL', $property));
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription(ResourceInterface $resource)
+    {
+        $description = [];
+        foreach ($this->getClassMetadata($resource)->getFieldNames() as $fieldName) {
+            if ($this->isPropertyEnabled($fieldName)) {
+                $description += $this->getFilterDescription($fieldName, self::PARAMETER_BEFORE);
+                $description += $this->getFilterDescription($fieldName, self::PARAMETER_AFTER);
```

Comment:
Wouldn't it be better to have something like : `dateProperty[after | before]` instead of 2 definitions ?